pyro/infer/inspect.py

```diff
@@ -276,17 +282,34 @@ def model(data):
             trace = poutine.trace(model).get_trace(*model_args, **model_kwargs)
 
     sample_sample = {}
+    sample_param = {}
     sample_dist = {}
+    param_constraint = {}
     plate_sample = defaultdict(list)
     observed = []
+
+    def _get_type_from_frozenname(frozen_name):
+        return trace.nodes[frozen_name]["type"]
+
     for name, site in trace.nodes.items():
+        if site["type"] == "param":
+            param_constraint[name] = site["kwargs"]["constraint"]
```

Comment:
Could we store a string representation of the constraint here, rather than the `Constraint` object?  For example:
```py
param_constraint[name] = str(site["kwargs"]["constraint"])
```
The reason is that the constraint object may contain tensors, and storing the tensor values in the dependencies dict may leak memory. This dict is intended to be pure-Python with no PyTorch data structures.

Reply:
Oh thanks for pointing out, I have corrected it.